src/main/java/games/strategy/engine/data/GameObjectStreamData.java

```diff
@@ -62,8 +62,9 @@ Named getReference(final GameData data) {
           return data.getProductionRuleList().getProductionRule(m_name);
         case PRODUCTIONFRONTIER:
           return data.getProductionFrontierList().getProductionFrontier(m_name);
+        default:
+          throw new IllegalStateException("Unknown type: " + m_type);
```

Comment:
I changed `this` to `m_type` in the exception message because it seemed like a more useful piece of information to report.